src/actions/queue/useQueue.ts

```diff
@@ -152,6 +156,20 @@ export class QueueManager {
 		});
 	}
 
+	/** Add a like to the message */
+	async addLike(queueMessage: Discord.Message | Discord.PartialMessage): Promise<void> {
+		const message = await queueMessage.fetch();
+
+		const currentLikeCount = fetchLikeCountFromString(message.content);
+		const newLikeCount = currentLikeCount + 1;
+
+		await this.queueStorage.setLikeCount(newLikeCount, queueMessage.id);
+
+		await editMessage(message, {
+			content: replaceLikeCountInString(newLikeCount, message.content)
```

Comment:
I think we should go with a declarative approach here, perhaps using `StringBuilder`. Rather than reading the string to edit it, which may be happening multiple times concurrently, let's instead build the correct final string based on what the database knows about the entry:
* `isDone` defines whether a strikethrough is present and embeds are suppressed
* `likeCount` defines the number of likes we display
* other attributes that I may have forgotten, or that we want to add later

Reply:
This is something I've wanted to do in `useQueue` for a while now. I'll build this function on `main`, so you have something to modify, rather than create from scratch.

Reply:
Alright, thanks ^^

Reply:
New code merged. LMK if that's useful to you